src/zeek.bif

```diff
@@ -5615,3 +5644,47 @@ function find_in_zeekpath%(p: string%): string
 
 	return zeek::make_intrusive<zeek::StringVal>(resolved.c_str());
 	%}
+
+## Returns a vector of strings containing the tags of all plugin components
+## that belong to the specified category.
+##
+## @param category
+##    A string representing the category of components to filter by.
+##    Valid categories include: "READER", "WRITER", "ANALYZER", "PACKET_ANALYZER",
+##    "FILE_ANALYZER", "IOSOURCE", "PKTSRC", "PKTDUMPER", "SESSION_ADAPTER",
+##    "CLUSTER_BACKEND", "EVENT_SERIALIZER", "LOG_SERIALIZER", "STORAGE_BACKEND",
+##    "STORAGE_SERIALIZER".
+##
+## @return
+##    A vector of strings (string_vec) with the tags of matching components.
+##    Returns null and raises an error if the category is unknown.
+function get_tags_by_category%(category: string%): string_vec
+	%{
+    std::string cat = category->CheckString();
```

Comment:
Total nit: `.bif` files are typically indented with tabs. I know that's different from the rest of the code. It annoys me too.